### PR TITLE
[password-hash-and-salt] Add typings

### PR DESCRIPTION
**Pull request by alitaheri**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If adding a new definition:
- [x] The package does not provide its own types, and you can not add them.
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, and `strictNullChecks` set to `true`.

### CODE REVIEW

File: types/password-hash-and-salt/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+interface Password {
+    hash(cb: (error: string, hash: string) => void): void;
+    verifyAgainst(hash: string, cb: (error: string, verified: boolean) => void): void;
+}
+
+declare function password(password: string): Password;
+
+declare const _: typeof password;
```
Comment (andy-ms) [inline types/password-hash-and-salt/index.d.ts]: Just `export = password` should do.

Comment (alitaheri) [inline types/password-hash-and-salt/index.d.ts]: I tried that initially, but this seems to be the only solution that allows:

```ts
import * as password from 'password-hash-and-salt';
```

Comment (andy-ms) [inline types/password-hash-and-salt/index.d.ts]: If the module uses `export = password` it's not technically correct to import with `import * as password` (one is a CommonJS export, the other is an ES6 import). There is a workaround though where adding `declare namespace password {}` (with no body) will allow that syntax.

**Comment by alitaheri** [comment]
@andy-ms I've done what you've asked. Please take another look. Thank you 👍

**Comment by andy-ms** [comment]
Thanks!

**Pull request merged**